src/dvbapi.cpp

```diff
@@ -678,8 +679,14 @@ int send_ecm(int filter_id, unsigned char *b, int len, void *opaque) {
 }
 
 int set_algo(SKey *k, int algo, int mode) {
-    if (algo == CA_ALGO_AES128 && mode == CA_MODE_CBC)
+    // Use CBC when mode indicates it
+    if (algo == CW_ALGO_AES128 && mode == CW_ALGO_MODE_CBC)
         algo = CA_ALGO_AES128_CBC;
+
+    // Treat ICAM ECM the same as normal ECM
+    if (algo == CW_ALGO_CSA_ALT)
+        algo = CA_ALGO_DVBCSA;
+
```

Comment:
@catalinii currently we're only changing the value of `algo` here if it differs between the value indicated by DVBAPI and the value we expect in our descramblers. If you want to I can make it explicit, that way we could also show a proper error message if we receive a DES key instead of `op not found for algo 1`.